internal/recorder/recorder.go

```diff
@@ -0,0 +1,52 @@
+package recorder
+
+import (
+	"context"
+	"unicode"
+
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclienttest"
+	clientv1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+type Config struct {
+	Component string
+	K8sClient k8sclient.Interface
+}
+
+type Recorder struct {
```

Comment:
what do you think about calling this recorder `K8sEventsRecorder` or something like that? since this specific implementation will create k8s events

Reply:
I coped the most of the things from aws-operator, but your suggestion looks great. I will change the name. Thanks!